client/src/main/scala/skuber/api/WatchSource.scala

```diff
@@ -0,0 +1,121 @@
+package skuber.api
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.stream.scaladsl.{Broadcast, Flow, GraphDSL, JsonFraming, Merge, Source}
+import akka.stream.{Materializer, SourceShape}
+import akka.util.ByteString
+import play.api.libs.json.{Format, JsError, JsSuccess, Json}
+import skuber.api.client._
+import skuber.{ObjectResource, ResourceDefinition}
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
+import scala.concurrent.duration._
+
+object WatchSource {
```

Comment:
The approach is based on [1] 

[1] https://github.com/akka/akka/issues/20121

Reply:
@pbarron I was the original author of that Akka feature request. Just in case: please note that the discussion continued after the issue was moved to the akka-http project. The latest comments can be found here: https://github.com/akka/akka-http/issues/63

👍 on this PR by the way.

Reply:
The conversation at https://github.com/akka/akka-http/issues/63 is very useful in relation to this PR. One point brought up by @agemooij in that conversation is that per Akka docs having long polling requests on a shared pool can starve other requests. By default skuber uses `Http().singleRequest` which uses a shared pool. While the invoker can be overridden, doing that is relatively low-level and undocumented.
I think for `watch` requests in general (including the existing methods as well as the new ones in this PR) we should by default use an unshared connection pool with a single connection, along the lines of https://github.com/agemooij/stream-experiments/blob/master/src/main/scala/scalapenos/experiments/streams/LongPollingHttpClient.scala.
Potentially this PR could implement the code for that default and then a later update to the existing watch methods could reuse it.

Reply:
It is a good point and something I noticed but decided not to tackle in the PR as it was a reasonable change to how the client works. I'll update the PR to include an _unshared connection pool with a single connection_ and we can go from there.